### PR TITLE
upbound: 0.34.0 -> 0.35.3

### PR DESCRIPTION
**Pull request by jljox**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jljox** [comment]
@lucperkins 
Hi Luc, when you have a moment, could you please review this PR? Your feedback would be greatly appreciated.

**Comment by jeanlucjox** [comment]
@NixOS/nixpkgs-vet

**Review by lucperkins — CHANGES_REQUESTED** [review]
Just two small nits

File: pkgs/by-name/up/upbound/package.nix
Comment (lucperkins) [inline pkgs/by-name/up/upbound/package.nix:75]: This needs to be an `unfree` license

**Review by lucperkins — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (lucperkins) [inline pkgs/top-level/all-packages.nix:18872]: Using `pkgs/by-name` means that you don't need to add anything to this `all-packages.nix` file. Instead, you should have two directories with a `package.nix`:

* `pkgs/by-name/up/upbound`
* `pkgs/by-name/up/upbound-main`

**Comment by jljox** [comment]
Hey @lucperkins ,
Do we have in the nix community some kind of `induction` for new maintainers?

File: pkgs/by-name/up/upbound/package.nix
Comment (lucperkins) [reply pkgs/by-name/up/upbound/package.nix:73]: Why was this marked as resolved? This needs to be an unfree license. As far as I can tell, the source repo for Upbound is no longer public.

File: pkgs/by-name/up/upbound/package.nix
Comment (lucperkins) [inline pkgs/by-name/up/upbound/package.nix:73]: Again, this needs to be an unfree license. If you're pulling pre-built binaries using `fetchurl` and the repo is not public, then this cannot be marked as having an Apache license.

**Review by lucperkins — CHANGES_REQUESTED** [review]

**Review by lucperkins — APPROVED** [review]

**Review by lucperkins — APPROVED** [review]

File: pkgs/top-level/all-packages.nix
Comment (FliegendeWurst) [inline pkgs/top-level/all-packages.nix:18871]: There is a merge conflict here.
You can avoid it by:
1. specifying version-channel to be "stable" by default
2. creating upbound-main in pkgs/by-name and overriding upbound there

File: pkgs/by-name/up/upbound-main/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/up/upbound-main/package.nix:5]:
```suggestion
{ upbound }:

upbound.override {
  version-channel = "main";
}
```

Comment (FliegendeWurst) [reply pkgs/by-name/up/upbound-main/package.nix:5]: Packages in pkgs/by-name must be isolated, they are not allowed to refer to outside files.

File: pkgs/top-level/all-packages.nix
Comment (jljox) [reply pkgs/top-level/all-packages.nix]: Done. Thank you for the feedback :)

File: pkgs/by-name/up/upbound/package-common.nix
Comment (Scrumplex) [reply pkgs/by-name/up/upbound/package-common.nix:59]: IMO this should just stay unset. There is no checkPhase for this package.

File: pkgs/by-name/up/upbound/package.nix
Comment (Scrumplex) [inline pkgs/by-name/up/upbound/package.nix:5]: I think we can just `mv pkgs/by-name/up/upbound/package-common.nix pkgs/by-name/up/upbound/package.nix` and it should work just as well without the need for another callPackage here.

File: pkgs/by-name/up/upbound/package-common.nix
Comment (jljox) [reply pkgs/by-name/up/upbound/package-common.nix]: Agreed, done. :)

File: pkgs/by-name/up/upbound/package-common.nix
Comment (jljox) [reply pkgs/by-name/up/upbound/package-common.nix]: OK 👍 cheers

File: pkgs/by-name/up/upbound/package.nix
Comment (jljox) [reply pkgs/by-name/up/upbound/package.nix]: Nice one. Done 👍

Comment (jljox) [reply pkgs/by-name/up/upbound/package.nix]: Cool. Done 😃

File: pkgs/by-name/up/upbound/sources-main.nix
Comment (jljox) [reply pkgs/by-name/up/upbound/sources-main.nix:1]: Could I do this in a subsequent PR? Once this one is merged, I'll look into it 😀

File: pkgs/by-name/up/upbound/sources-stable.nix
Comment (jljox) [reply pkgs/by-name/up/upbound/sources-stable.nix:1]: Could I do this in a subsequent PR? Once this one is merged, I'll look into it 😀

File: pkgs/by-name/up/upbound-main/package.nix
Comment (Scrumplex) [inline pkgs/by-name/up/upbound-main/package.nix:4]:
```suggestion
{
  upbound,
}:
upbound.override {
```

Comment (jljox) [reply pkgs/by-name/up/upbound-main/package.nix]: Done 👍

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**
